oxenc/endian.h

```diff
@@ -40,75 +55,75 @@ concept endian_swappable_integer =
 /// Byte swaps an integer value unconditionally.  You usually want to use one of the other
 /// endian-aware functions rather than this.
 template <endian_swappable_integer T>
-void byteswap_inplace(T& val) {
+constexpr void byteswap_inplace(T& val) {
     if constexpr (sizeof(T) == 2)
-        val = static_cast<T>(bswap_16(static_cast<uint16_t>(val)));
+        val = bit_cast<T>(bswap_16(bit_cast<uint16_t>(val)));
```

Comment:
This still seems problematic because we have no guarantee that `bswap_N` is constexpr.  Under gcc/clang it is defined to `__builtin_bswap{N}` which apparently is, but for a general purpose library like this we at least try to support other compilers such as MSVC and non-gcc/clang on Linux, which will have `bswap_N` defined as something else.

There is an alternative, though, in the "possible implementation" of std::byteswap (https://en.cppreference.com/w/cpp/numeric/byteswap):
```C++
#ifdef __cpp_lib_byteswap
#include <byteswap>
#else
#include <algorithm>
#endif

// ...

namespace oxen {
#ifdef __cpp_lib_byteswap
using std::byteswap;
#else
// ... cppreference's possible implementation of byteswap ...
#endif

// and then update remaining code to use `byteswap`
```

Reply:
It would be really great if byteswap worked with c++20 but your point is excellent and I made a change w.r.t. it!

Reply:
It would be really great if byteswap worked with c++20 but your point is excellent and I made a change w.r.t. it!